### PR TITLE
Adds integration with albums external api

### PR DESCRIPTION
**Pull request by anaturrillo**
## Summary

I implemented the integration with an external api tha allows to get all albums, one album by id and an album's photos.

## Trello Card

[https://trello.com/c/FqObnnBu/36-integraci%C3%B3n-con-api-de-albumes](https://trello.com/c/FqObnnBu/36-integraci%C3%B3n-con-api-de-albumes)

### CODE REVIEW

File: app/services/get_albums.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const axios = require('axios');
+
+const endpoint = 'https://jsonplaceholder.typicode.com/albums';
```
Comment (mpicollo-wolox) [inline app/services/get_albums.js]: Make this url a config variable

File: app/routes.js [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
+const { albums, photos } = require('./controllers/albums');
+
 exports.init = app => {
   app.get('/health', healthCheck);
+  app.get('/albums', albums);
+  app.get('/albums/:id', albums);
+  app.get('/albums/:id/photos', photos);
   // app.get('/endpoint/get/path', [], controller.methodGET);
   // app.put('/endpoint/put/path', [], controller.methodPUT);
   // app.post('/endpoint/post/path', [], controller.methodPOST);
```
Comment (mpicollo-wolox) [inline app/routes.js]: Remove this comments

File: app/controllers/albums.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const { getAlbums, getAlbum } = require('../services/get_albums');
+const { getPhotos } = require('../services/get_photos');
+
+exports.albums = (req, res, next) => {
+  const albumId = req.params.id;
+  const respond = albumId ? getAlbum : getAlbums;
```
Comment (mpicollo-wolox) [inline app/controllers/albums.js]: Improve name of `respond`

File: app/controllers/albums.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  return respond(albumId)
+    .then(response => res.json(response.data))
+    .catch(next);
+};
+
+exports.photos = (req, res, next) => {
+  const albumId = req.params.id;
+  return getPhotos(albumId)
+    .then(response => res.json(response.data))
```
Comment (mpicollo-wolox) [inline app/controllers/albums.js]: `.data` should be resolved by service, in that way if you change axios per request-promise, controllers don't have any changes

File: app/routes.js [hunk 5]
```diff
@@ -1,7 +1,12 @@
 // const controller = require('./controllers/controller');
 const { healthCheck } = require('./controllers/healthCheck');
+const { albums, photos } = require('./controllers/albums');
```
Comment (mpicollo-wolox) [inline app/routes.js]: Add some declarativity to controller functions, like `getAlbums`, `getPhotos`, etc

File: app/services/get_albums.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+
+exports.getAlbums = () => axios.get(endpoint);
```
Comment (mpicollo-wolox) [inline app/services/get_albums.js]: Add some catch here for map some API error, if you don't do that an ugly 5xx would be sent to client

File: app/controllers/albums.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+const { getAlbums, getAlbum } = require('../services/get_albums');
+const { getPhotos } = require('../services/get_photos');
```
Comment (mpicollo-wolox) [inline app/controllers/albums.js]: It seems to much one service per action (like `create_albums`, `update_albums`), services are external APIs (in most of the cases), makes sense have all this functions in one service. I don't see advantages of split this in differents files.

**Comment by anaturrillo** [comment]
I believe I've addressed every comment. I though think some things will become more "general" when the project gets bigger (like error and response handling).

File: app/services/externalServices.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+const formatData = a => a.data;
+
+const externalApiError = e => {
+  const status = e.response.status || 500;
+  const message = errorMessages[status.toString()] || errorMessages.general;
+  return {
+    status,
+    message
+  };
```
Comment (mpicollo-wolox) [inline app/services/externalServices.js]: See `middlewares/errors`

File: config/externalEndpoints.js [hunk 9]
```diff
@@ -0,0 +1,4 @@
+exports.externalEndpoints = {
+  albums: 'https://jsonplaceholder.typicode.com/albums',
+  photos: 'https://jsonplaceholder.typicode.com/photos'
```
Comment (mpicollo-wolox) [inline config/externalEndpoints.js]: Check `config/index.js` for this values, make this an environment variable

File: app/helpers/index.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+exports.errorMessages = {
+  404: 'We did not find what you where looking for',
+  400: 'Please check you request data',
+  503: 'Looks like something went wrong on our side',
+  general: 'We are unable to process your request'
```
Comment (mpicollo-wolox) [inline app/helpers/index.js]: Check `app/errors`

File: app/controllers/albums.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
+const { getAllAlbums, getAlbumById, getPhotosByAlbum } = require('../services/externalServices');
+
+const sendResponse = res => response => {
+  res.status(response.status || 200);
+  res.json(response);
```
Comment (mpicollo-wolox) [inline app/controllers/albums.js]: I don't see this abstraction like too necessary, could be more simple resolve this in controller directly like `res.send(response)` (default status code is 200 and response is sent as JSON)

File: package.json [hunk 12]
```diff
@@ -54,14 +54,15 @@
     }
   },
   "dependencies": {
+    "axios": "^0.19.0",
```
Comment (mpicollo-wolox) [inline package.json]: Missing `package-lock.json` update, i think CI is failing for that

File: app/services/externalServices.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
+const axios = require('axios');
+const { albumsEndpoint, photosEndpoint } = require('../../config').common.externalApi;
+const { externalApiError } = require('../errors');
+
+const formatData = a => a.data;
```
Comment (mpicollo-wolox) [inline app/services/externalServices.js]: Use some meaning name for this, like `response => response.data`

**Pull request merged**